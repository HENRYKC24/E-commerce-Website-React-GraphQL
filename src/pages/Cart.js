import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import OneCartItem from '../components/CartItem';
import styles from './cart.module.css';
import { removeFromCart } from '../redux/products/products';

class Cart extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      cart: [],
    };
  }

  componentDidMount() {
    const { cart } = this.props;
    this.setState({ cart });
  }

  render() {
    const {
      cartContainer,
      cartHeading,
      cartList,
      noProduct,
      noProductContainer,
    } = styles;

    const { cart } = this.state;

    return (
      <section className={cartContainer}>
        {cart.length > 0 ? (
          <>
            <p className={cartHeading}>Cart</p>
            <ul className={cartList}>
              {cart.map((each) => (
                <OneCartItem
                  key={Math.random()}
                  removeThisItemFromCart={() => {
                    const { dispatch } = this.props;
                    dispatch(removeFromCart(each.id));
                    setTimeout(() => {
                      const { cart } = this.props;
                      this.setState({ cart });
                    }, 50);
                  }}
                  itemId={each.id}
                  data={each}
                />
              ))}
            </ul>
          </>
        ) : (
          <div className={noProductContainer}>
            <p className={noProduct}>No product in the cart.</p>
          </div>
        )}
      </section>
    );
  }
}

Cart.propTypes = {
  cart: PropTypes.instanceOf(Array).isRequired,
  dispatch: PropTypes.func.isRequired,
};

function mapStateToProps({ state }) {
  return state;
}

export default connect(mapStateToProps)(Cart);
