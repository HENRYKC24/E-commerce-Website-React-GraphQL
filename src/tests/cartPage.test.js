import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Cart from '../pages/Cart';
import store from '../redux/configureStore';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

it('Login renders the UI as expected', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Cart />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
