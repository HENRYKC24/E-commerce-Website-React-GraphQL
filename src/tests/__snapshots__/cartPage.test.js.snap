// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders the UI as expected 1`] = `
<section
  className="cartContainer"
>
  <div
    className="noProductContainer"
  >
    <p
      className="noProduct"
    >
      No product in the cart.
    </p>
  </div>
</section>
`;
